Source/JavaScriptCore/API/glib/JSCValue.cpp

```diff
@@ -2145,3 +2145,61 @@ char* jsc_value_to_json(JSCValue* value, unsigned indent)
 
     return json;
 }
+
+/**
+ * JSCExecutor:
+ * @resolve: #JSCValue function to call to resolve the promise
+ * @reject: #JSCValue function to call to reject the promise
+ * @user_data: user data
+ *
+ * Function invoked at promise initialization
+ *
+ * It is called like a js function, so exceptions raised will not be propagated
+ * to the context, but handled by the promise causing a rejection.
+ * @resolve and @reject can be reffed for later use to handle async tasks completion.
```

Comment:
The doc comments need more work. This isn't sufficient to explain what a JSCExecutor is or why you would want to use one. I see it's really simple -- just a function that you pass to `jsc_value_new_promise()` -- so might as well say that here.